test/matrix_generator.cc

```diff
@@ -1921,17 +1947,88 @@ void generate_matrix(
         case TestMatrixType::poev:
         case TestMatrixType::heev:
         default:
-            slate_error("Not implemented yet");
-            throw std::exception();  // not implemented
+            snprintf( msg, sizeof( msg ), "'%s' not yet implemented",
+                      params.kind().c_str() );
+            throw std::runtime_error( msg );
     }
 
     if (! (type == TestMatrixType::rand  ||
            type == TestMatrixType::rands ||
            type == TestMatrixType::randn ||
            type == TestMatrixType::randb) && dominant) {
         // make diagonally dominant; strict unless diagonal has zeros
-        slate_error("Not implemented yet");
-        throw std::exception();  // not implemented
+        snprintf( msg, sizeof( msg ), "in '%s', dominant not yet implemented",
+                  params.kind().c_str() );
+        throw std::runtime_error( msg );
+    }
+
```

Comment:
@neil-lindquist This is the new implementation for trapezoid storage matrices, including Hermitian.